Source/ProjectNL/GAS/Ability/Active/Default/ComboAttack/AnimNotify/EnableCollisionNotifyState.cpp

```diff
@@ -0,0 +1,249 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+
+#include "EnableCollisionNotifyState.h"
+#include "AbilitySystemBlueprintLibrary.h"
+#include "ProjectNL/Character/BaseCharacter.h"
+#include "ProjectNL/Component/EquipComponent/EquipComponent.h"
+#include "ProjectNL/Weapon/BaseWeapon.h"
+
+void UEnableCollisionNotifyState::NotifyBegin(USkeletalMeshComponent* MeshComp, UAnimSequenceBase* Animation, float TotalDuration)
+{
+    UE_LOG(LogTemp, Log, TEXT("Notify Begin"));
+
+    // 애니메이션 시작 시 이전 프레임의 소켓 위치 초기화
+    PrevStartLocation = FVector::ZeroVector;
+    PrevEndLocation = FVector::ZeroVector;
+}
+
+void UEnableCollisionNotifyState::NotifyTick(USkeletalMeshComponent* MeshComp, UAnimSequenceBase* Animation, float FrameDeltaTime)
+{
+    if (AActor* Owner = MeshComp->GetOwner())
+    {
+        MakeTriangleTrace(Owner);
+    }
+}
+
+void UEnableCollisionNotifyState::NotifyEnd(USkeletalMeshComponent* MeshComp, UAnimSequenceBase* Animation)
+{
+    if (AActor* Owner = MeshComp->GetOwner())
+    {
+      {
+            HitActors.Reset();
+        }
+    }
+    UE_LOG(LogTemp, Log, TEXT("Notify End"));
+}
+FVector BezierCurve(const FVector& P0, const FVector& P1, const FVector& P2, float t)
+{
+    float u = 1 - t;
+    float tt = t * t;
+    float uu = u * u;
+
+    FVector point = uu * P0; // (1 - t)^2 * P0
+    point += 2 * u * t * P1; // 2(1 - t)t * P1
+    point += tt * P2;        // t^2 * P2
+
+    return point;
+}
+FVector SplineCurve(const FVector& P0, const FVector& P1, const FVector& P2, float t)
+{
+    // t는 곡선에서의 위치를 나타내며, 0 <= t <= 1
+    float t2 = t * t;
+    float oneMinusT = 1.0f - t;
+
+    // 2차 스플라인 곡선 공식
+    FVector point = (oneMinusT * oneMinusT) * P0;     // (1 - t)^2 * P0
+    point += 2 * oneMinusT * t * P1;                 // 2 * (1 - t) * t * P1
+    point += t2 * P2;                                // t^2 * P2
+
+    return point;
+}
+void UEnableCollisionNotifyState::MakeTriangleTrace(AActor* Owner)
+{
+    
+    ABaseCharacter* BaseCharacter= Cast<ABaseCharacter>(Owner);
+    UEquipComponent* EquipComponent{};
+    if(BaseCharacter)
+    {
+        EquipComponent= BaseCharacter->GetEquipComponent();
+    }
+    ABaseWeapon* MainWeapon{};
+    if(EquipComponent)
+        MainWeapon=EquipComponent->GetMainWeapon();
```

Comment:
```suggestion
  if (ABaseCharacter* BaseCharacter= Cast<ABaseCharacter>(Owner)) {
      if (UEquipComponent* EquipComponent = BaseCharacter->GetEquipComponent()) {
        ABaseWeapon* MainWeapon = EquipComponent->GetMainWeapon();
        .
        .
        .
      }
  }
```

이런 패턴이 더 좋아보여요. 굳이 매번 새로운 빈 객체 인스턴스를 만들고 새롭게 할당해주는 것보다 처음부터 if문에서 할당해주고 예외처리하는 것이 더 깔끔해 보이는데 어떻게 생각하세요?

Reply:
이렇게 되면은 if문으로 너무 둘러싸일 것 같아서 이렇게 진행했습니다.저는 개인적으로 if문에 넣어서 할당하는 거 보다 빼서 할당하는 게 좀 더 나중에 재사용 할 때 좋은 거 같더라구요.